coderd/prometheusmetrics/prometheusmetrics.go

```diff
@@ -328,29 +328,42 @@ func Agents(ctx context.Context, logger slog.Logger, registerer prometheus.Regis
 					templateVersionName = "unknown"
 				}
 
-				user, err := db.GetUserByID(ctx, workspace.OwnerID)
-				if err != nil {
-					logger.Error(ctx, "can't get user from the database", slog.F("user_id", workspace.OwnerID), slog.Error(err))
-					agentsGauge.WithLabelValues(VectorOperationAdd, 0, user.Username, workspace.Name, templateName, templateVersionName)
-					continue
+				username := workspace.OwnerUsername
+				// This should never be possible, but we'll guard against it just in case.
+				// 1. The owner_id field on the workspaces table is a reference to IDs in the users table and has a `NOT NULL` constraint
```

Comment:
Can we not just add a foreign key constraint on this field? Seems like an odd omission.
Along with that, a constraint to enforce the minimum length of the username would add a lot more integrity.

This would likely render the rest of this discussion moot as this situation would not be possible.
I do appreciate the defensiveness of this code though, thanks @cstyan 👍

Reply:
Good idea 👍 that PR should block this one.